common/logger/src/diem_logger.rs

```diff
@@ -263,6 +275,7 @@ pub struct DiemLogger {
     sender: Option<SyncSender<LoggerServiceEvent>>,
     printer: Option<Box<dyn Writer>>,
     filter: RwLock<DiemFilter>,
+    custom_format: Option<Box<dyn Fn(&LogEntry) -> Result<String, fmt::Error> + Send + Sync>>,
```

Comment:
How about instead of special casing the format function we conditionally set the closure based on if the user wants the default formatter or not. That'll eliminate some of the special casing code down below.

Reply:
Got it. So instead of being an `Option`, rename to `formatter` and have it use the default formatter or the custom formatter based on whtat the user selected on the builder? That sounds good

Reply:
exactly!